Source/Mosa.Compiler.MosaTypeSystem.CLR/Metadata/ClrMetadata.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) MOSA Project. Licensed under the New BSD License.
+
+namespace Mosa.Compiler.MosaTypeSystem.CLR.Metadata
+{
+	internal class ClrMetadata : IMetadata
+	{
+		private readonly ClrModuleLoader moduleLoader;
+
+		public ClrMetadata(ClrModuleLoader loader)
+		{
+			moduleLoader = loader;
+		}
+
+		public TypeSystem? TypeSystem { get; private set; }
+
+		public ITypeSystemController? Controller { get; private set; }
+
+		public ClrMetadataCache? Cache { get; private set; }
+
+		public ClrMetadataLoader? Loader { get; private set; }
```

Comment:
What's the difference between `Loader` and `moduleLoader`? They look like they're for the same purpose but use two separate instances

Reply:
The ClrModuleLoader (``moduleLoader``) loads all ``ModuleDefMD``s into a list, while the ClrMetadataLoader (``Loader``) loads the types, methods, etc.. for a module. It can be a ``ModuleDef`` (dnlib) or a ``MosaModule``, depending on how early it's called.